strider/server.py

```diff
@@ -328,17 +330,21 @@ async def sync_query(
     qid = str(uuid.uuid4())[:8]
     try:
         LOGGER.info(f"[{qid}] Starting sync query")
-        query_results = await asyncio.wait_for(
-            lookup(query_dict, qid), timeout=max_process_time
+        # get max timeout
+        timeout = max(
+            max_process_time, query_dict.get("parameters", {}).get("timeout_seconds", 0)
```

Comment:
I forget the complete details for fastAPI query param parsing. Is it possible for timeout_seconds to end up here as a string or some other non-numeric type.

Reply:
This isn't a query param, it's in a dedicated `parameters` field in the request body on the same level as `message`. And there's no type checking for `timeout_seconds`, so someone could pass whatever they wanted. `parameters` isn't part of the TRAPI spec, so I highly doubt anyone outside of ranking-agent devs are going to even know that this exists.

Reply:
Added type checks